zerver/lib/upload.py

```diff
@@ -194,13 +194,33 @@ def get_total_uploads_size_for_user(user: UserProfile) -> int:
         total_quota = 0
     return total_quota
 
+def get_total_uploads_size_for_realm(realm: Realm) -> int:
+    uploads = Attachment.objects.filter(realm=realm)
+    total_quota = uploads.aggregate(Sum('size'))['size__sum']
+
+    # In case realm has no uploads
+    if (total_quota is None):
+        total_quota = 0
+    return total_quota
+
 def within_user_upload_quota(user: UserProfile, uploaded_file_size: int) -> bool:
     total_quota = get_total_uploads_size_for_user(user)
     if (total_quota + uploaded_file_size > user.quota):
         return False
     else:
         return True
 
+def within_realm_upload_quota(realm: Realm, uploaded_file_size: int) -> bool:
+    if realm.uploads_quota is None:
```

Comment:
I would consider writing this as 
```
def check_upload_within_quota(realm, uploaded_file_size) -> None:
    if realm.uploads_quota is None
        return
    used_space = currently_used_upload_space(realm)
    if used_space + uploaded_file_size > realm.uploads_quota:
        raise RealmUploadLimitException
```

And then catching and handling the exception in the caller. This seems like the right semantics since in the end this is  going to raise an error for the user.